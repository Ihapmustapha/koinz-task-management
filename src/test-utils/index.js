export { default as MockedProvider } from "./mockedProvider";
