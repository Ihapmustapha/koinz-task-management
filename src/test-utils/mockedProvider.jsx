import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { configureStore } from "../store";

const MockedProvider = ({ initialState, children }) => (
  <Provider store={configureStore(initialState)}>{children}</Provider>
);

MockedProvider.propTypes = {
  initialState: PropTypes.shape({}).isRequired,
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
};

export default MockedProvider;
