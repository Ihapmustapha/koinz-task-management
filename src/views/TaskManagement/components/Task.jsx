import React from "react";
// prop-types
import PropTypes from "prop-types";
// react-beautiful-dnd
import { Draggable } from "react-beautiful-dnd";
// components
import { Paper, Grid, Typography, withStyles } from "@material-ui/core";
// styles
import styles from "../styles";

const Task = ({ task, index, onClick, classes }) => {
  const { id, description, history, taskStatus } = task;
  return (
    <Draggable
      draggableId={id}
      index={index}
      isDragDisabled={taskStatus && taskStatus === "done"}
    >
      {(provided, snapshot) => (
        <Paper
          className={classes.taskCard}
          {...provided.dragHandleProps}
          {...provided.draggableProps}
          innerRef={provided.innerRef}
        >
          <Grid
            item
            container
            direction="column"
            justify="flex-start"
            alignItems="flex-start"
            onClick={onClick}
            id={id}
          >
            <Grid item xs={12} lg={12}>
              {description && (
                <Typography variant="caption" component="p">
                  {description}
                </Typography>
              )}
            </Grid>
          </Grid>
        </Paper>
      )}
    </Draggable>
  );
};

Task.propTypes = {
  task: PropTypes.shape({
    id: PropTypes.string,
    description: PropTypes.string,
    history: PropTypes.shape,
    taskStatus: PropTypes.string,
  }).isRequired,
  index: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    taskCard: PropTypes.string,
  }).isRequired,
};
export default withStyles(styles)(Task);
