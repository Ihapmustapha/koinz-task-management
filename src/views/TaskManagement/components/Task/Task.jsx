import React from "react";
// prop-types
import PropTypes from "prop-types";
// react-beautiful-dnd
import { Draggable } from "react-beautiful-dnd";
// components
import { Paper, Grid, Typography, withStyles, Fab } from "@material-ui/core";
// styles
import styles from "../../styles";
// icons
import ViewIcon from "../../../../assets/view.svg";

const Task = ({ task, index, onClick, classes }) => {
  const { id, description, taskStatus } = task;
  return (
    <Draggable
      draggableId={id}
      index={index}
      isDragDisabled={taskStatus && taskStatus === "done"}
    >
      {(provided) => (
        <Paper
          className={classes.taskCard}
          {...provided.dragHandleProps}
          {...provided.draggableProps}
          innerRef={provided.innerRef}
        >
          <Grid
            item
            container
            direction="column"
            justify="flex-start"
            alignItems="flex-start"
          >
            <Grid item xs={12} lg={12}>
              {description && (
                <Typography variant="caption" component="p">
                  {description}
                </Typography>
              )}
            </Grid>
            <Grid
              item
              container
              direction="row"
              justify="flex-end"
              alignItems="flex-end"
              xs={12}
              lg={12}
            >
              <Fab
                size="small"
                onClick={onClick(id)}
                variant="extended"
                color="inherit"
                style={{ boxShadow: "none" }}
              >
                <img alt="view icon" src={ViewIcon} style={{ width: "15px" }} />
              </Fab>
            </Grid>
          </Grid>
        </Paper>
      )}
    </Draggable>
  );
};

Task.propTypes = {
  task: PropTypes.shape({
    id: PropTypes.string,
    description: PropTypes.string,
    history: PropTypes.shape({}),
    taskStatus: PropTypes.string,
  }).isRequired,
  index: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    taskCard: PropTypes.string,
  }).isRequired,
};
export default withStyles(styles)(Task);
