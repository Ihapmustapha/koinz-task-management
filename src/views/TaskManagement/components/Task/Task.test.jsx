import React from "react";
import { mount } from "enzyme";

import Task from "./Task";

import { MockedProvider } from "../../../../test-utils";

jest.mock("react-beautiful-dnd", () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {}
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {}
    ),
  DragDropContext: ({ children }) => children,
}));

describe("<Task /> with no props", () => {
  const getContainer = (props) =>
    mount(
      <MockedProvider initialState={{ modalState: { isOpen: false } }}>
        <Task {...props} />
      </MockedProvider>
    );

  const container = getContainer({
    index: 0,
    onClick: (id) => {},
    classes: {},
    task: {
      id: "testTask",
      description: "test task",
      history: {},
      taskStatus: "todo",
    },
  });

  container.update();

  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  it("should contain a material ui paper as a card", () => {
    expect(container.find("[data-testid='task-paper-test-id']").exists()).toBe(
      true
    );
  });

  it("should have typography to render task description", () => {
    expect(
      container.find("[data-testid='desc-typography-test-id']").exists()
    ).toBe(true);
  });

  it("should have typography to render task description", () => {
    expect(container.find("[data-testid='view-task-fab']").exists()).toBe(true);
  });
});
