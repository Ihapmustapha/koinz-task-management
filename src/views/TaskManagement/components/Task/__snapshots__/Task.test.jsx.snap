// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> with no props should match the snapshot 1`] = `"<div class=\\"MuiPaper-root Task-taskCard-7 MuiPaper-elevation1 MuiPaper-rounded\\" id=\\"task-paper\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12\\"><p class=\\"MuiTypography-root MuiTypography-caption\\" id=\\"desc-typography-test-id\\">test task</p></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-12 MuiGrid-grid-lg-12\\"><button class=\\"MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeSmall MuiFab-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" style=\\"box-shadow: none;\\" id=\\"view-task-fab\\"><span class=\\"MuiFab-label\\"><img alt=\\"view icon\\" src=\\"view.svg\\" style=\\"width: 15px;\\"></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div>"`;
